scripts/setupTypeScriptRollup.js

```diff
@@ -0,0 +1,264 @@
+	'@types/node': '^14.0.27',
+	'@types/polka': '^0.5.1',
+	'svelte-check': '^1.0.0',
+	'svelte-preprocess': '^4.0.0',
+	'typescript': '^3.9.3',
+	'tslib': '^2.0.0'
+});
+
+// Add script for checking
+packageJSON.scripts = Object.assign(packageJSON.scripts, {
+	validate: 'svelte-check --ignore src/node_modules/@sapper'
+});
+
+// Write the package JSON
+fs.writeFileSync(pkgJSONPath, JSON.stringify(packageJSON, null, '  '));
+
+scanFolderAndReplace(path.join(projectRoot, 'src'));
+
+// replace js filenames to ts
+function scanFolderAndReplace(dir) {
+	const elements = fs.readdirSync(dir, { withFileTypes: true });
+
+	for (let i = 0; i < elements.length; i++) {
+		if (elements[i].isDirectory()) {
+			scanFolderAndReplace(path.join(dir, elements[i].name));
+		} else if (elements[i].name.match(/^((?!json).)*js$/)) {
+			fs.renameSync(
+				path.join(dir, elements[i].name),
+				path.join(dir, elements[i].name).replace('.js', '.ts')
+			);
+		}
+	}
+}
+
+// Switch the *.svelte file to use TS
+[
+	{
+		view: 'components/Nav',
+		vars: [{ name: 'segment', type: 'string' }]
+	},
+	{
+		view: 'routes/index'
+	},
+	{
+		view: 'routes/about'
+	},
+	{
+		view: 'routes/_layout',
+		vars: [{ name: 'segment', type: 'string' }]
+	},
+	{
+		view: 'routes/_error',
+		vars: [
+			{ name: 'status', type: 'number' },
+			{ name: 'error', type: 'Error' }
+		]
+	},
+	{
+		view: 'routes/blog/index',
+		vars: [
+			{ name: 'posts', type: '{ slug: string; title: string, html: any }[]' }
+		],
+		contextModule: [
+			{
+				js: '.then(r => r.json())',
+				ts: '.then((r: { json: () => any; }) => r.json())'
+			},
+			{
+				js: '.then(posts => {',
+				ts: '.then((posts: { slug: string; title: string, html: any }[]) => {'
+			}
+		]
+	},
+	{
+		view: 'routes/blog/[slug]',
+		vars: [{ name: 'post', type: '{ slug: string; title: string, html: any }' }]
+	}
+].forEach(({ view, vars, contextModule }) => {
+	const svelteFilePath = path.join(projectRoot, 'src', `${view}.svelte`);
+	let file = fs.readFileSync(svelteFilePath, 'utf8');
+
+	file = file.replace(/(?:<script)(( .*?)*?)>/gm, '<script$1 lang="ts">');
+
+	if (vars) {
+		vars.forEach(({ name, type }) => {
+			file = file.replace(
+				`export let ${name};`,
+				`export let ${name}: ${type};`
+			);
+		});
+	}
+
+	if (contextModule) {
+		contextModule.forEach(({ js, ts }) => {
+			file = file.replace(`${js}`, `${ts}`);
+		});
+	}
+
+	fs.writeFileSync(svelteFilePath, file);
+});
+
+// Edit rollup config
+const rollupConfigPath = path.join(projectRoot, 'rollup.config.js');
+let rollupConfig = fs.readFileSync(rollupConfigPath, 'utf8');
+
+// Edit imports
+rollupConfig = rollupConfig.replace(
+	`'rollup-plugin-terser';`,
+	`'rollup-plugin-terser';
+import sveltePreprocess from 'svelte-preprocess';
+import typescript from '@rollup/plugin-typescript';`
+);
+
+// Edit inputs
+rollupConfig = rollupConfig.replace(
+	`onwarn(warning);`,
+	`(warning.code === 'THIS_IS_UNDEFINED') ||\n\tonwarn(warning);`
```

Comment:
why do we need this? is it a workaround for https://github.com/sveltejs/sapper/pull/1468 not being merged yet?

Reply:
Yes, this way we can deliver a completely clean sapper-template with TypeScript and remove this once the PR is finished.